doc/plot.md

```diff
@@ -15,79 +13,131 @@ kernelspec:
 
 ```{code-cell} ipython3
-%pip install matplotlib --quiet
+from pathlib import Path
+from urllib.request import urlretrieve
+
+if not Path("yellow_tripdata_2021-01.parquet").is_file():
+    urlretrieve("https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2021-01.parquet",
+                "yellow_tripdata_2021-01.parquet")
 ```
 
+### Setup
+
+```{note}
+`%sqlplot` requires `matplotlib`: `pip install matplotlib` and this example requires
+duckdb-engine: `pip install duckdb-engine`
+```
+
++++
+
+Load the extension and connect to an in-memory DuckDB database:
+
 ```{code-cell} ipython3
 %load_ext sql
 ```
 
-Connect to an in-memory SQLite database.
-
 ```{code-cell} ipython3
-%sql sqlite://
+%sql duckdb://
 ```
 
-## Line
+We'll be using a sample dataset that contains historical taxi data from NYC:
+
++++
+
+### Data preview
 
 ```{code-cell} ipython3
-%%sql sqlite://
-CREATE TABLE points (x, y);
-INSERT INTO points VALUES (0, 0);
-INSERT INTO points VALUES (1, 1.5);
-INSERT INTO points VALUES (2, 3);
-INSERT INTO points VALUES (3, 3);
+%%sql
+SELECT * FROM "yellow_tripdata_2021-01.parquet" LIMIT 3
 ```
 
 ```{code-cell} ipython3
-points = %sql SELECT x, y FROM points
-points.plot()
+%%sql
+SELECT COUNT(*) FROM "yellow_tripdata_2021-01.parquet"
 ```
 
-## Bar
+## Boxplot
 
-+++
+```{note}
+To use `%sqlplot boxplot`, your SQL engine must support:
+
+`percentile_disc(...) WITHIN GROUP (ORDER BY ...)`
+
+[Snowflake](https://docs.snowflake.com/en/sql-reference/functions/percentile_disc.html),
+[Postgres](https://www.postgresql.org/docs/9.4/functions-aggregate.html),
+[DuckDB](https://duckdb.org/docs/sql/aggregates), and others support this.
+```
 
-*Note: sample data from the TIOBE index.*
+To create a boxplot, call `%sqlplot boxplot`, and pass the name of the table, and the column you want to plot. Since we're using DuckDB for this example, the table is the path to the parquet file.
 
 ```{code-cell} ipython3
-%%sql sqlite://
-CREATE TABLE languages (name, rating, change);
-INSERT INTO languages VALUES ('Python', 14.44, 2.48);
-INSERT INTO languages VALUES ('C', 13.13, 1.50);
-INSERT INTO languages VALUES ('Java', 11.59, 0.40);
-INSERT INTO languages VALUES ('C++', 10.00, 1.98);
+%sqlplot boxplot --table yellow_tripdata_2021-01.parquet --column trip_distance
 ```
 
+There are many outliers in the data, let's find the 90th percentile to use it as cutoff value for better visualization:
+
 ```{code-cell} ipython3
-change = %sql SELECT name, change FROM languages
-change.bar()
+%%sql
+SELECT percentile_disc(0.90) WITHIN GROUP (ORDER BY trip_distance),
+FROM 'yellow_tripdata_2021-01.parquet'
 ```
 
-## Pie
+Now, let's create a query that filters by the 90th percentile. Note that we're using the `--save`, and `--no-execute` functions. This tells JupySQL to store the query, but *skips execution*. We'll reference it in our next plotting call.
```

Comment:
If I'm using --save it'll always execute? It's kind of the opposite of what I expected.

Reply:
Yes, `--save` stores the query and executes by default.

Reply:
Is this legacy behavior? We might want to discuss/see if users use the no-execute flag a lot and inverse the behavior

Reply:
this is a new feature we released a few months ago. I agree, might be better to switch the behavior.